app/services/google_scraping_service.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'httparty'
+require 'nokogiri'
+
+class GoogleScrapingService
+  attr_reader :keyword_id, :keyword
+
+  def initialize(keyword_id, keyword)
+    @keyword_id = keyword_id
+    @keyword = keyword
+  end
+
+  def call!
+    scrap_from_google
```

Comment:
Should we update the Keyword status to `:in_progress` before `scrap_from_goole` as since the keyword is failed (for the first time for example) the status is still in `failed` and when it retries here, the status still as `failed` till this job is done, and the status will change to `completed` but in case the `scrap_from_google` take a long time to finish then the Keyword still display as `failed`.

```
def call!
    Keyword.update(
      keyword_id,
      status: :in_progress)
    scrap_from_google
    store_result
end
```

Reply:
Job doesn't update status to `failed` even job fail at the first time [Ref](https://edgeapi.rubyonrails.org/classes/ActiveJob/Exceptions/ClassMethods.html#method-i-retry_on). It will update status to `failed` when retry amount reach the maximum config.